### DISCUSSION TITLE
Now that Svelte 5 is coming, will these carbon components for svelte be compatible with it?

### DISCUSSION THREAD

**Discussion by SrikanthVH** [carbon-design-system/carbon-components-svelte#1908] [General]
As per the latest announcements, Svelte 5 comes with lot of breaking changes. Will the already existing carbon components for svelte continue to work or is there any plan to incorporate the changes to make it compatible with Svelte 5 ?

**Comment by theetrain** [comment]
This is partly covered in https://github.com/carbon-design-system/carbon-components-svelte/issues/1877

To my knowledge, `carbon-components-svelte` should already be compatible with Svelte 5, but we won't know for sure until there's a stable release.

**Comment by simonwiles** [comment]
Doesn't seem to work for me.

Starting a new project (with `"@sveltejs/kit": "^2.0.0"` and `"svelte": "^5.0.0-next.1"`), importing anything from `'carbon-components-svelte'` results in:

```
[plugin:vite-plugin-svelte] /path/to/project/node_modules/.pnpm/carbon-components-svelte@0.82.11/node_modules/carbon-components-svelte/src/Link/Link.svelte:53:6 <div> is invalid inside <p>
```

Same code works fine with `"svelte": "^4.2.7"`.

**Reply by theetrain** [reply]
That's interesting; HTML validation errors will fail in Svelte 5. We should remove the `<p>` since Carbon React doesn't use that, either.

Let's continue in this issue: https://github.com/carbon-design-system/carbon-components-svelte/issues/1924

**Reply by simonwiles** [reply]
Thanks -- changing the `<p ... />` tag to `<a aria-disabled="true" ... />` as suggested in #1924 and replacing a couple of uses of `$` as a variable name in a couple of components has got me started using `carbon-components-svelte` with Svelte 5.  I'll post back here if (when?) I come to regret this decision... :)

**Reply by theetrain** [reply]
Fix was released in [v0.83.0](https://github.com/carbon-design-system/carbon-components-svelte/releases/tag/v0.83.0).

**Comment by metonym** [comment]
The aim for this library is to be compatible with Svelte 3, 4, and 5 (non-Runes mode).

Parent issue to track compatibility issues and bugs: https://github.com/carbon-design-system/carbon-components-svelte/issues/2012
